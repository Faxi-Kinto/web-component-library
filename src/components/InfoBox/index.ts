import InfoBox from './InfoBox.component';

export default InfoBox;
