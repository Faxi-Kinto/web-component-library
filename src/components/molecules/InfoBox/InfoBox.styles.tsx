import styled from 'styled-components';
import { theme } from '../../../config';
import { size, positionAbsolute, padding } from '../../../styles/basics';
import { pxToRem } from '../../../styles/basics';

export const Container = styled.div`
  position: relative;
  border: ${pxToRem('0.5px')} solid ${theme.pallet.DISABLED};
  background-color: ${theme.pallet.WHITE};
  ${padding(pxToRem('18px'), pxToRem('30px'), pxToRem('16px'))};

  &.info-box--has-icon {
    ${padding(
      pxToRem('18px'),
      pxToRem('30px'),
      pxToRem('16px'),
      pxToRem('50px')
    )};
  }

  .info-box__icon {
    color: ${theme.pallet.LIGHT_BLUE_2};
    ${size(pxToRem('22px'))};
    ${positionAbsolute(pxToRem('10px'), '', '', pxToRem('13px'))};
  }
`;
