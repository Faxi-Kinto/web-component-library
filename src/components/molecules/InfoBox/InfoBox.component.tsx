import React from 'react';
import * as Styled from './InfoBox.styles';

/**
 * @name InfoBox
 *
 * [Insert Component Description]
 *
 * @returns {JSX}
 */

export type InfoBoxProps = {
  /** @param {ReactNode} icon
   *  @description Prop that represents icon,
   *  you can send either JSX object or React Component
   */
  children: React.ReactNode;
};

const InfoBox: React.FC<InfoBoxProps> = (props: InfoBoxProps): JSX.Element => {
  const { children } = props;

  return <Styled.Container>{children}</Styled.Container>;
};

export default InfoBox;
