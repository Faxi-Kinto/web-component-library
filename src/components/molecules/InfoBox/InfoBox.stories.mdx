import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import InfoBox from './';

<Meta title="Core|Molecules/InfoBox" />

# InfoBox Component

Component used to brief user with the information.

## Implementations

#### Default

<Preview>
  <Story name="default">
    <InfoBox>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
      tempor incididunt ut labore et dolore magna aliqua.
    </InfoBox>
  </Story>
</Preview>

#### Without an Icon

<Preview>
  <Story name="Without icon">
    <InfoBox>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
      tempor incididunt ut labore et dolore magna aliqua.
    </InfoBox>
  </Story>
</Preview>

## Props

<Props of={InfoBox} />

## Notes

- Prop `body` can either be passed as a prop `<InfoBox body={<p>hello</p>}/>` or
  passed as `children` => `<InfoBox><p>hello</p></InfoBox/>`. Use whichever you find
  more convenient.
