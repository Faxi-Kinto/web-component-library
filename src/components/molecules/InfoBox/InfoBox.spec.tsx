import React from 'react';
import renderer from 'react-test-renderer';
import InfoBox from './InfoBox.component';

test('InfoBox renders correctly', () => {
  const component = renderer.create(
    <InfoBox>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
        tempor incididunt ut labore et dolore magna aliqua.
      </p>
    </InfoBox>
  );

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
