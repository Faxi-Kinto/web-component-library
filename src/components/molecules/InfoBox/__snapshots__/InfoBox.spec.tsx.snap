// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBox renders correctly 1`] = `
.c0 {
  position: relative;
  padding: 1.125rem 1.875rem 1rem 1.875rem;
}

.c0.info-box--has-icon {
  padding: 1.125rem 1.875rem 1rem 3.125rem;
}

.c0 .info-box__icon {
  width: 1.375rem;
  height: 1.375rem;
  position: absolute;
  top: 0.625rem;
  left: 0.8125rem;
}

<div
  className="c0"
>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
  </p>
</div>
`;
